### DISCUSSION TITLE
Polar align utility

### DISCUSSION THREAD

**Discussion by gregdev** [artyom-beilis/OpenLiveStacker#101] [Ideas]
This app is amazing, thanks for all the effort so far!

Something that would be very useful (to me, at least) is a polar alignment utility, similar to that offered by Sharpcap - you roughly align your mount to the NCP/SCP, it plate solves, then you rotate your mount about RA (manually in my case - I use a Star Adventurer 2i), it plate solves again and then gives you the adjustments required in left/right/up/down movements needed to get things aligned to the pole.

There is a Python app - https://github.com/ThemosTsikas/PhotoPolarAlign - that does this, which might assist with the required logic.

I'd love to help but am more of a web guy - I haven't touched C++ since uni - so might be slightly out of my depth.

Cheers!

**Comment by artyom-beilis** [comment]
The problem is I don't actually own equatorial mount. I use my AZ GTi in Alt-Az mode - since it is way faster and easier.

But this is something frequently requested indeed and I'll get to it at some point.

**Comment by gregdev** [comment]
@artyom-beilis you are an absolute legend for implementing this! Now I no longer need to lug my laptop with me when I go out hunting stars. Unfortunately there are clouds forecast for the next several days but I am keen to give this a test when I get clear skies.

Thanks again!

**Comment by gregdev** [comment]
I finally had a couple of hours under clear dark-ish skies recently and tried out the polar alignment using my ASI120MC-S + SVBONY SV165. At first I struggled with plate solving ("status: failed to find solution") but this may have been my dodgy initial rough alignment - I used SharpCap on my laptop to polar align (using the SharpSolve built-in plate solver) and then tried out OpenLiveStacker again and success, it plate solved and agreed with SharpCap about the state of my alignment.

I'll give this another go whenever I next get the opportunity!

**Comment by artyom-beilis** [comment]
Cool.

I didn't manage to test it against reference and you have given me the validation I needed!

My testing was:

- the result was consistent across multiple measurements
- error getting lower as I align.
- tracking become way better 

So thanks for the report!
